sqlvm-provisioning-csp/SQLServerVMDeployment.json

```diff
@@ -0,0 +1,238 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
```

Comment:
Please remove default value empty strings from parameters that don't need them. It's an optional parameter.